hathor/transaction/base_transaction.py

```diff
@@ -879,6 +881,7 @@ def clone(self) -> 'BaseTransaction':
         """
         new_tx = self.create_from_struct(self.get_struct())
         if hasattr(self, '_metadata'):
+            assert self._metadata is not None  # FIXME: is this actually true or do we have to check if not None
```

Comment:
I think if we have `_metadata` set, it will never be `None`, but I don't see harm in having this assert.

Reply:
If this is the case we could annotate it with `_metadata: TransactionMetadata`, this would allow us to confidently remove this assertion, but I'd guess it would need some work to get around all the typing issues that would be revealed.

But I'm OK leaving this for a greater full mypy-strict refactor.

Reply:
@obiyankenobi  The assert is required for mypy.

And I agree with @jansegre, we could do it in other ways, but this requires fixing the typing issues, which I agree that could be done in another refactor.